mountpoint-s3/src/data_cache/in_memory_data_cache.rs

```diff
@@ -3,13 +3,15 @@
 use std::collections::HashMap;
 use std::default::Default;
 
+use async_trait::async_trait;
+
 use super::{BlockIndex, ChecksummedBytes, DataCache, DataCacheError, DataCacheResult};
 use crate::object::ObjectId;
-use crate::sync::RwLock;
+use crate::sync::AsyncRwLock;
 
 /// Simple in-memory (RAM) implementation of [DataCache]. Recommended for use in testing only.
 pub struct InMemoryDataCache {
-    data: RwLock<HashMap<ObjectId, HashMap<BlockIndex, ChecksummedBytes>>>,
+    data: AsyncRwLock<HashMap<ObjectId, HashMap<BlockIndex, ChecksummedBytes>>>,
```

Comment:
not sure if this change adds any value [if we don't hold the lock across await points ](https://docs.rs/async-lock/latest/async_lock/#relationship-with-stdsync)

Reply:
You are correct! Reverted.